docs/new-spec.md

```diff
@@ -0,0 +1,347 @@
+
+# Introduction
+
+Guac is an Ethereum single hop payment channel client for guac Bi-Directional Payment
+Channel contract. This light client is able to send and verify channel opening, updating, and ending
+transactions.
+
+# Problem
+
+Guac_rs used to implement Ethcalate (aka Connectix) contract client https://github.com/althea-mesh/simple-bidirectional-erc20-channel/ to implement the payment logic. That contract code turns out to have more issues than expected and we decided to move back to the original guac contract located at https://github.com/althea-mesh/guac/.
+
+With the new implementation of the payment logic we want to focus on the reusability and make guac_rs generic enough to be able to be reused in other projects as well, while at the same it should ship ready to use implementation that other projects can use straight away.
+
+# Components
+
+There are various components that are used to communicate between parties off-chain (i.e. through HTTP) and on-chain (i.e. Web3 protocol).
+
+## Transport layer
+
+A set of functions that allows a set of instances of Guac_rs to communicate with each other over a reliable transport layer such as HTTP.
+
+_Open question: What about making it a proper RESTful API i.e. POST /channel, DELETE /channel/id for closing, POST /channel/:id/created?_
+
+### Proposing a channel
+
+`POST /propose`
+
+Proposes a channel from party A to party B.
+
+Request parameters:
+
+- `channel_id` - A random 32 bytes string that matches the regexp `[a-f0-9]{32}`.
```

Comment:
this is a 128 bit random number (each hex digit is 4 bits), when ethereum uses 256 bit integers. Did you mean to do {64}?